pkg/kapp/diff/diff_resources.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package diff
+
+import (
+	"sort"
+
+	ctlconf "github.com/vmware-tanzu/carvel-kapp/pkg/kapp/config"
+	ctlres "github.com/vmware-tanzu/carvel-kapp/pkg/kapp/resources"
+)
+
+type DiffResources struct {
+	ExistingResources, NewResources versionedResources
+	ExistingResourcesGrouped        map[string][]ctlres.Resource
+
+	newRs []ctlres.Resource
```

Comment:
saving newRs is only a workaround for RenewableResources, I would prefer a different approach, but did not want to spend to much time here for the moment :)